### PR TITLE
customGear v2

### PR DESCRIPTION
**Pull request by McDiod**
supersedes #46

### CODE REVIEW

File: functions/customgear/fn_getCustomGearOptions.sqf [hunk 1]
```diff
@@ -0,0 +1,27 @@
+#include "component.hpp"
+
+params [["_unit", objNull],["_loadoutHash", []], ["_ignoreCurrentLoadout", false]];
```
Comment (Salbei2500) [inline functions/customgear/fn_getCustomGearOptions.sqf]:
```suggestion
params [["_unit", objNull], ["_loadoutHash", []], ["_ignoreCurrentLoadout", false]];
```

File: functions/customgear/fn_onCustomGearTabButton.sqf [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+_ctrlListBoxPosition set [1, [-safeZoneH, (ctrlPosition _button) select 1] select (count _availableOptions > 0)];
+_ctrlListBox ctrlSetPosition _ctrlListBoxPosition;
+_ctrlListBox ctrlSetFade 0;
+_ctrlListBox ctrlCommit 0;
+{
+    private _itemClassname = _x;
+    private _parentClass = "";
+    {
+        if (isClass (configFile >> _x >> _itemClassname)) exitWith {_parentClass = _x};
+    } forEach ["CfgWeapons","CfgMagazines","CfgVehicles"];
```
Comment (Salbei2500) [inline functions/customgear/fn_onCustomGearTabButton.sqf]:
```suggestion
    } forEach ["CfgWeapons", "CfgMagazines", "CfgVehicles"];
```

File: functions/customgear/fn_onCustomGearTabButton.sqf [hunk 3]
```diff
@@ -0,0 +1,107 @@
+#include "component.hpp"
+
+params ["_button", "_isLeftSide"];
+
+private _display = ctrlParent _button;
+private _unit = _display getVariable [QGVAR(unit), objNull];
+
+// move selection indicator to selected tab
+private _ctrlTabSelected = _display getVariable [[QGVAR(ctrlTabSelectedRight),QGVAR(ctrlTabSelectedLeft)] select _isLeftSide, controlNull];
```
Comment (Salbei2500) [inline functions/customgear/fn_onCustomGearTabButton.sqf]:
```suggestion
private _ctrlTabSelected = _display getVariable [[QGVAR(ctrlTabSelectedRight), QGVAR(ctrlTabSelectedLeft)] select _isLeftSide, controlNull];
```

File: functions/customgear/fn_updateCamera.sqf [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+] find _hashKey;
+
+if (_keyID < 0) exitWith {};
+
+private _selectionData = [
+    ["Pelvis", 0, [0.4,2.4,0.3]],
+    ["Spine3", -0.1, [0,1.4,0.1]],
+    ["Spine3", 0, [-1,-1.2,0]],
+    ["proxy:\A3\Characters_F\Proxies\weapon.001", [-0.1, 0.1] select (GVAR(curSelectedWeaponID) == 0), [[0.8,-0.8,0],[0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 0)],
+    ["proxy:\A3\Characters_F\Proxies\launcher.001", 0, [[-1.7,-0.1,0],[0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 1)],
```
Comment (Salbei2500) [inline functions/customgear/fn_updateCamera.sqf]:
```suggestion
    ["proxy:\A3\Characters_F\Proxies\launcher.001", 0, [[-1.7,-0.1,0], [0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 1)],
```

File: functions/customgear/fn_updateCamera.sqf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    "radio"
+] find _hashKey;
+
+if (_keyID < 0) exitWith {};
+
+private _selectionData = [
+    ["Pelvis", 0, [0.4,2.4,0.3]],
+    ["Spine3", -0.1, [0,1.4,0.1]],
+    ["Spine3", 0, [-1,-1.2,0]],
+    ["proxy:\A3\Characters_F\Proxies\weapon.001", [-0.1, 0.1] select (GVAR(curSelectedWeaponID) == 0), [[0.8,-0.8,0],[0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 0)],
```
Comment (Salbei2500) [inline functions/customgear/fn_updateCamera.sqf]:
```suggestion
    ["proxy:\A3\Characters_F\Proxies\weapon.001", [-0.1, 0.1] select (GVAR(curSelectedWeaponID) == 0), [[0.8,-0.8,0], [0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 0)],
```

File: functions/customgear/fn_updateCamera.sqf [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+    ["proxy:\A3\Characters_F\Proxies\launcher.001", 0, [[-1.7,-0.1,0],[0.8,0.8,0]] select (GVAR(curSelectedWeaponID) == 1)],
+    ["proxy:\A3\Characters_F\Proxies\pistol.001", 0, [[0.4,2.4,0.3],[0.5,0.5,0]] select (GVAR(curSelectedWeaponID) == 2)],
```
Comment (Salbei2500) [inline functions/customgear/fn_updateCamera.sqf]:
```suggestion
    ["proxy:\A3\Characters_F\Proxies\pistol.001", 0, [[0.4,2.4,0.3], [0.5,0.5,0]] select (GVAR(curSelectedWeaponID) == 2)],
```

File: functions/general/fn_randomizeLoadout.sqf [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+
+params [["_loadoutHash", []]];
+
+{
+    private _value = [_loadoutHash, _x] call CBA_fnc_hashGet;
+    if (_value isEqualType []) then {
+        if (count _value == 0) then {_value = ""} else {_value = selectRandom _value};
+        [_loadoutHash, _x, _value] call CBA_fnc_hashSet;
+    };
+} forEach [
```
Comment (Salbei2500) [inline functions/general/fn_randomizeLoadout.sqf:13]: Shouldn't the new macro be used here?

**Review by Salbei2500 — CHANGES_REQUESTED** [review]
See above

File: functions/general/fn_randomizeLoadout.sqf (hunk 7 above)
Comment (McDiod) [reply functions/general/fn_randomizeLoadout.sqf:13]: No, the macro is specific to the customGear component. It doesn't include `xxxWeaponMagazine` and `xxxWeaponUnderbarrelMagazine`.

Comment (Salbei2500) [reply functions/general/fn_randomizeLoadout.sqf:13]: Ah okay

**Pull request merged**